### PR TITLE
Ruby on Rails, Databases: Improve lessons per Style Guide

### PR DESCRIPTION
**Pull request by scheals**
Complete the following REQUIRED checkboxes:

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`

Complete the following checkboxes ONLY IF they are applicable to your PR. You can complete them later if they are not currently applicable:
-   [x] I have previewed all lesson files included in this PR with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure the Markdown content is formatted correctly
-   [x] I have ensured all lesson files included in this PR follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

<hr>

**1. Because:**
The lessons in this section weren't adhering by the Style Guide.

**2. This PR:**
Attempts to make them adhere to the Style Guide to my best ability.

**3. Additional Information:**
Possibly decide whether `_` should be used for italics - I followed the example of previous `_(__note:__)_` occurrences but `*` is used in Markdown as well, so might want to clear this up just like with `-` for unordered lists.

### CODE REVIEW

File: databases/databases/databases_and_sql.md [hunk 1]
```diff
@@ -18,52 +18,17 @@ SQL is one of those topics that's been stored away in dusty old technical manual
...
-* `MAX`
-* `SUM`
-
-#### Other Stuff:
-* What are Indexes good for?
-* What's the difference between `WHERE` and `HAVING`?
+### Lesson Overview
+This section contains a general overview of topics that you will learn in this lesson.
+
+- What the Primary Key is.
```
Comment (ChargrilledChook) [inline databases/databases/databases_and_sql.md:24]: I think "What a Primary Key is" reads a bit more naturally. Thoughts?

File: databases/databases/databases_and_sql.md [hunk 2]
```diff
@@ -107,14 +72,14 @@ A close cousin of `SELECT`, for if you only want unique values of a column, is `
 
 If you want to get all the posts created by a given user, you need to tell SQL which columns it should use to zip the tables together with the `ON` clause. Perform the "zipping" with the `JOIN` command.  But wait, if you mash two tables together where the data doesn't perfectly match up (e.g. there are multiple posts for one user), which rows do you actually keep?  There are four different possibilities:
 
-Note: the "left" table is the original table (the one that the `FROM` clause was `ON`), e.g. "users" in examples below.
+_(__note:__ the "left" table is the original table (the one that the `FROM` clause was `ON`), e.g. "users" in examples below.)_
```
Comment (ChargrilledChook) [inline databases/databases/databases_and_sql.md:75]: I haven't seen this before (`_(__note:__`), is this a shortcut to get the info box? Everytime I've used that I've done it with plain HTML

File: databases/databases/databases_and_sql.md [hunk 3]
```diff
@@ -147,7 +112,7 @@ The last nifty trick is if you want to only display a subset of your data.  In a
   HAVING posts_written >= 10;
 ~~~
 
-Try going back to [the W3 example](http://www.w3schools.com/sql/trysql.asp?filename=trysql_select_groupby) and joining the `Customers` and the `Orders` tables to get the number of orders in each country and adding the line `HAVING COUNT(*) > 10;` after `GROUP BY` (and delete the extra semicolon in the previous line).
+Try going back to [the W3 Schools' example](http://www.w3schools.com/sql/trysql.asp?filename=trysql_select_groupby) and joining the `Customers` and the `Orders` tables to get the number of orders in each country and adding the line `HAVING COUNT(*) > 10;` after `GROUP BY` (and delete the extra semicolon in the previous line).
```
Comment (ChargrilledChook) [inline databases/databases/databases_and_sql.md:115]: Assuming W3 School is singular, we would want `School's` here

File: databases/databases/databases.md [hunk 4]
```diff
@@ -6,41 +6,43 @@ Databases are kind of hidden in the back of the web application so people treat
...
+This section contains a general overview of topics that you will learn in this lesson.
 
-* What is a database?
-* What are relational databases?
-* How are relational databases different from XML?
-* What is SQL?
-* What is SQL used for?
-* How do you get all the records from a table in SQL?
-* How do you insert a record in SQL?
+- What a database is.
```
Comment (ChargrilledChook) [inline databases/databases/databases.md:12]: Same comments re: list spacing from #24773

**Review by ChargrilledChook — COMMENTED** [review]
Nice stuff again @scheals, again just some minor comments.

File: databases/databases/databases_and_sql.md (hunk 2 above)
Comment (scheals) [reply databases/databases/databases_and_sql.md:75]: So in #24704 and #24705 I added notes with such formatting, I believe I took the idea from here:
https://github.com/TheOdinProject/curriculum/blob/main/foundations/installations/installations.md?plain=1#L120
Note that just after it there's a note with different formatting and it is a lil' broken. I unify how these notes look in #24776 in an attempt to make those notes more uniform and readable by making them start from a newline.

Possibly something to introduce in the Style Guide if you guys figure out which way you want to go.

Comment (scheals) [reply databases/databases/databases_and_sql.md:75]: Oh also, no, this is not the info box. The `_` before and after the `(`, `)` make the text italic, while `__note:__` makes it bold.

See `Additional information` of this PR:
3. Additional Information:
Possibly decide whether `_` should be used for italics - I followed the example of previous _(__note:__)_ occurrences but `*` is used in Markdown as well, so might want to clear this up just like with `-` for unordered lists.

File: databases/databases/databases_and_sql.md (hunk 3 above)
Comment (scheals) [reply databases/databases/databases_and_sql.md:115]: Nope, they are indeed called W3Schools. https://www.w3schools.com/about/default.asp But I might be daft, so check me on this 😆

File: databases/databases/databases_and_sql.md (hunk 1 above)
Comment (scheals) [reply databases/databases/databases_and_sql.md:24]: True. Thanks for noting!

File: databases/databases/databases.md (hunk 4 above)
Comment (scheals) [reply databases/databases/databases.md]: Should be fixed now!

File: databases/databases/project_sql_zoo.md [hunk 5]
```diff
@@ -5,14 +5,15 @@ SQL Zoo is one of the few resources online that actually lets you build and run
...
-  2. Before you move on, we would like your feedback [here](https://docs.google.com/forms/d/e/1FAIpQLSenvMG6WFbOOEap_biQOwqfbH-j-xsf5Eyv4ir2Rx5FsYSecQ/viewform?usp=sf_link). Getting user(you) feedback is important so we can continue to improve the curriculum and get an idea of your experience.
+  1.  Go to [SQL Zoo](https://sqlzoo.net/) and do Tutorials 0-9 listed under the "Tutorial Section" and the quizzes listed at the end of each.  The first is the ["Select" tutorial](https://sqlzoo.net/wiki/SELECT_basics).
+      -   Make sure the dropdown on the upper right of the main page for "Engine" says "MySQL" (the default).  Large results will be cut off and not all rows or columns shown, so the "answers" may not look 100% correct.
+  2.  Before you move on, we would like your feedback - [please fill this form](https://docs.google.com/forms/d/e/1FAIpQLSenvMG6WFbOOEap_biQOwqfbH-j-xsf5Eyv4ir2Rx5FsYSecQ/viewform?usp=sf_link). Getting user (you) feedback is important so we can continue to improve the curriculum and get an idea of your experience.
```
Comment (scheals) [inline databases/databases/project_sql_zoo.md:10]: @ChargrilledChook 
I know you haven't mentioned this bit but I changed the prompt for filling out the form to be more in like with how linking should be done, I know that there are other cases like this and could make them match this - does it make sense?

File: databases/databases/databases_and_sql.md (hunk 3 above)
Comment (ChargrilledChook) [reply databases/databases/databases_and_sql.md:115]: Seems a bit odd to me, but you're absolutely correct :+1:

**Review by wise-king-sullyman — APPROVED** [review]
Looks great to me, thanks!

I'm holding off on merging for the moment though since I see you pinged @ChargrilledChook for a question.

File: databases/databases/project_sql_zoo.md (hunk 5 above)
Comment (ChargrilledChook) [reply databases/databases/project_sql_zoo.md:10]: @scheals I compared and don't have strong feelings about it either way. Your edit probably improves readability slightly, but it's not major.

Contributor's choice :slightly_smiling_face:

**Review by ChargrilledChook — APPROVED** [review]

File: databases/databases/project_sql_zoo.md (hunk 5 above)
Comment (scheals) [reply databases/databases/project_sql_zoo.md:10]: Then I'll keep my edit :D

**Comment by scheals** [comment]
I think it's good to go now? Hope that's how I was meant to resolve the conflict, no idea how to squash this merge commit 😆

**Comment by wise-king-sullyman** [comment]
I wouldn't want you to squash the merge commit anyways haha. Thanks!

**Pull request merged**